oai_agents/agents/rl.py

```diff
@@ -295,31 +295,43 @@ def train_agents(self, total_train_timesteps, exp_name=None):
         self.log_details(experiment_name, total_train_timesteps)
 
         if self.checkpoint_rate is not None:
-            self.ck_list = []
-            path, tag = self.save_agents(tag=f'ck_{len(self.ck_list)}')
-            self.ck_list.append(({k: 0 for k in self.args.layout_names}, path, tag))
-
+            if self.args.resume:
+                import os
+
+                path = self.args.base_dir / 'agent_models' / experiment_name
+
+                ckpts = [name for name in os.listdir(path) if name.startswith("ck")]
+                ckpts_nums = [int(c.split('_')[1]) for c in ckpts]
+                sorted_idxs = np.argsort(ckpts_nums)
+                ckpts = [ckpts[i] for i in sorted_idxs]
+                self.ck_list = [({k: 0 for k in self.args.layout_names}, path, ck) for ck in ckpts]
```

Comment:
@avv-va How should this dict be filled in when loading from checkpoints?

Reply:
Also the "tag" has the reward in it, is that correct or should I strip the reward?